Add distributed category to TORCH_LOGS

wconstab opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #103288
* __->__ #103287

This lets users run `TORCH_LOGS="+distributed" python myscript.py` and enable additional logging output for the distributed module.

wconstab committed:
Add distributed category to TORCH_LOGS

[ghstack-poisoned]

ezyang commented:
This doesn't seem right. Why is distributed an artifact logger? I would expect it to be a regular logger rooted at torch.distributed

wconstab committed:
Update on "Add distributed category to TORCH_LOGS"


This lets users run `TORCH_LOGS="+distributed" python myscript.py` and enable additional logging output for the distributed module. 


[ghstack-poisoned]

wconstab commented:
good catch. i copypasted the wrong thing :( this looks better, i confirmed it works, but please double check in case something is subtly wrong (also next PR) @ezyang

ZainRizvi commented:
@pytorchbot revert -m "Breaks trunk" -c landrace

See https://hud.pytorch.org/pytorch/pytorch/commit/0b252aebb2e46c0dc5585ec6d296832a308f563b for details

It's a weird failure, but looks like test is expecting the log to behave differently
